lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider.rb

```diff
@@ -61,7 +61,14 @@ def validate_upload_document(lighthouse_document)
   # @param file_body [String]
   def submit_upload_document(lighthouse_document, file_body)
     log_upload_attempt
-    api_response = BenefitsDocuments::Form526::UploadSupplementalDocumentService.call(file_body, lighthouse_document)
+
+    begin
+      api_response = BenefitsDocuments::Form526::UploadSupplementalDocumentService.call(file_body, lighthouse_document)
+    rescue => e
+      log_upload_failure(e)
+      raise e
+    end
+
     handle_lighthouse_response(api_response)
   end
 
```

Comment:
Minor suggestion: it might help to rename `handle_lighthouse_response` to `handle_lighthouse_success_response`